src/ecpool.erl

```diff
@@ -81,13 +81,36 @@ pool_spec(ChildId, Pool, Mod, Opts) ->
 %% @doc Start the pool sup.
 -spec(start_pool(pool_name(), atom(), [option()]) -> {ok, pid()} | {error, term()}).
 start_pool(Pool, Mod, Opts) ->
-    ecpool_pool_sup:start_link(Pool, Mod, Opts).
+    %% See start_sup_pool/3 for an explanation of InitialConnectResponseRef
+    InitialConnectResponseRef = make_ref(),
+    {ok, Pid} = OkResponse = ecpool_pool_sup:start_link(Pool, Mod, Opts, {self(), InitialConnectResponseRef}),
```

Comment:
maybe spawn a pid to collect the responses because this pid (and the reference) is stored by supervisor as child-spec.

Reply:
Ok, yes, I think this will work.

Reply:
I have changed the code to use a special purpose collector process: https://github.com/emqx/ecpool/pull/50/commits/bf64597eb9cc4765c2be0f40b56f97a69d2e2ded

Reply:
Changed again to use process alias instead: [3fceff6](https://github.com/emqx/ecpool/pull/50/commits/3fceff69511a2bff6b3712b86edb2ca30d0ad730)